streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -211,28 +207,44 @@ public synchronized void start() {
      * and then wait for them to join.
      *
      * @throws IllegalStateException if process has not started yet
+     * @param timeout   how long to wait for {@link StreamThread}s to shutdown
+     * @param timeUnit  unit of time used for timeout
      */
-    public synchronized void close() {
+    public synchronized void close(final long timeout, final TimeUnit timeUnit) {
```

Comment:
I'm not sure about exposing these params to the user. Other close() calls don't have timeouts. I'm not sure how I'm supposed to reason about the timeout as a user. Is 10 seconds sufficient? Or 60? Or 1? Could we have a reasonable internal default instead?